pkg/controller/release.go

```diff
@@ -164,20 +164,29 @@ func (e *helmExternal) Observe(ctx context.Context, mg resource.Managed) (manage
 	}
 
 	cr.Status.AtProvider = generateObservation(rel)
-	u, err := isUpToDate(ctx, e.localKube, &cr.Spec.ForProvider, rel, cr.Status)
+	s, err := isUpToDate(ctx, e.localKube, &cr.Spec.ForProvider, rel, cr.Status)
 	if err != nil {
 		return managed.ExternalObservation{}, errors.Wrap(err, errFailedToCheckIfUpToDate)
 	}
+	cr.Status.Synced = s
+	shouldCallUpdate := !s
+
+	if cr.Status.AtProvider.State == release.StatusDeployed && s {
+		cr.Status.Failed = 0
+	} else if cr.Status.AtProvider.State == release.StatusFailed &&
+		cr.Spec.RollbackLimit != nil && cr.Status.Failed < *cr.Spec.RollbackLimit {
+		shouldCallUpdate = true
+	}
 
 	return managed.ExternalObservation{
 		ResourceExists:   true,
-		ResourceUpToDate: u,
+		ResourceUpToDate: !shouldCallUpdate,
 	}, nil
 }
 
-type chartAction func(release string, chart *chart.Chart, vals map[string]interface{}, patches []ktype.Patch) (*release.Release, error)
+type deployFunc func(release string, chart *chart.Chart, vals map[string]interface{}, patches []ktype.Patch) (*release.Release, error)
```

Comment:
helm refers to these (install/upgrade etc) as `actions`
https://github.com/helm/helm/tree/d6708667da5c79ab4a093ea7473e96b93be5f1f7/pkg/action

Reply:
`chartAction` sounds like we are doing something on charts (e.g. pull, pack, unpack) and but it is in fact a deployment action. 
So, I am fine with action too, renamed as `deployAction`.